homeassistant/components/google_pubsub/__init__.py

```diff
@@ -0,0 +1,97 @@
+"""
+Support for Google Cloud Pub/Sub.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/google_pubsub/
+"""
+import datetime
+import json
+import logging
+import os
+from typing import Any, Dict
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    EVENT_STATE_CHANGED, STATE_UNAVAILABLE, STATE_UNKNOWN)
+from homeassistant.core import Event, HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entityfilter import FILTER_SCHEMA
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['google-cloud-pubsub==0.39.1']
+
+DOMAIN = 'google_pubsub'
+
+CONF_PROJECT_ID = 'project_id'
+CONF_TOPIC_NAME = 'topic_name'
+CONF_SERVICE_PRINCIPAL = 'credentials_json'
+CONF_FILTER = 'filter'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_PROJECT_ID): cv.string,
+        vol.Required(CONF_TOPIC_NAME): cv.string,
+        vol.Required(CONF_SERVICE_PRINCIPAL): cv.string,
+        vol.Required(CONF_FILTER): FILTER_SCHEMA
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass: HomeAssistant, yaml_config: Dict[str, Any]):
+    """Activate Google Pub/Sub component."""
+    from google.cloud import pubsub_v1  # pylint: disable=E0611
+
+    config = yaml_config.get(DOMAIN, {})
+    project_id = config[CONF_PROJECT_ID]
+    topic_name = config[CONF_TOPIC_NAME]
+    service_principal_path = os.path.join(hass.config.config_dir,
+                                          config[CONF_SERVICE_PRINCIPAL])
+
+    if not os.path.isfile(service_principal_path):
+        _LOGGER.error("Path to credentials file cannot be found")
+        return False
+
+    os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = service_principal_path
+
+    entities_filter = config[CONF_FILTER]
+
+    publisher = pubsub_v1.PublisherClient()
+    topic_path = publisher.topic_path(project_id,  # pylint: disable=E1101
```

Comment:
Why do we need to disable this pylint error?

Reply:
Pylint complains (incorrectly) about `topic_path` being not a member of `PublisherClient`

Reply:
Looks like library is using some advanced ways of building the client api.